### PR TITLE
Use enhanced switch statement

### PR DESCRIPTION
**Pull request by mx123**
## Description

Use enhanced switch statement

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java [hunk 1]
```diff
@@ -1673,22 +1673,17 @@ public Optional<ConnectorTableExecuteHandle> getTableHandleForExecute(
             Map<String, Object> executeProperties,
             RetryMode retryMode)
     {
-        DeltaLakeTableHandle tableHandle = (DeltaLakeTableHandle) connectorTableHandle;
-
-        DeltaLakeTableProcedureId procedureId;
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:1678]: Keep this part as it was before.
There can appear IAE while initializing the `procedureId`.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 2]
```diff
@@ -392,30 +392,33 @@ public IcebergTableHandle getTableHandle(
...
+        if (table.snapshot(snapshotId) == null) {
+            throw new TrinoException(INVALID_ARGUMENTS, "Iceberg snapshot ID does not exists: " + snapshotId);
         }
-        throw new TrinoException(NOT_SUPPORTED, "Version pointer type is not supported: " + version.getPointerType());
+        return snapshotId;
+    }
+
+    private static long getTemporalSnapshotIdFromVersion(Table table, ConnectorTableVersion version, io.trino.spi.type.Type versionType)
+    {
+        if (versionType instanceof TimestampWithTimeZoneType) {
```
Comment (findinpath) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:415]: Use here enhanced `instanceof` to avoid the extra casting in the subsequent line.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 3]
```diff
@@ -948,14 +951,10 @@ public Optional<ConnectorTableExecuteHandle> getTableHandleForExecute(
         }
 
         switch (procedureId) {
```
Comment (findinpath) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:953]: use `return` and drop the subsequent IAE - the compiler will do in this case the exhaustiveness check in the switch.

Comment (mx123) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java]: ok. done.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java (hunk 2 above)
Comment (mx123) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java]: done.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java (hunk 1 above)
Comment (mx123) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:1678]: reverted

File: lib/trino-geospatial-toolkit/src/main/java/io/trino/geospatial/serde/GeometrySerde.java [hunk 4]
```diff
@@ -286,21 +265,12 @@ public static Envelope deserializeEnvelope(Slice shape)
...
-            case POLYGON:
-            case MULTI_POLYGON:
-                return getSimpleGeometryEnvelope(input, length);
-            case GEOMETRY_COLLECTION:
-                return getGeometryCollectionOverallEnvelope(input);
-            case ENVELOPE:
-                return readEnvelope(input);
-        }
-        throw new IllegalArgumentException("Unexpected type: " + type);
+        return switch (requireNonNull(type, "geometrySerializationType is null")) {
```
Comment (findinpath) [inline lib/trino-geospatial-toolkit/src/main/java/io/trino/geospatial/serde/GeometrySerde.java:268]: why `requireNonNull` ? it wasn't there before your change.

btw

```
requireNonNull(type, "geometrySerializationType is null")
```

should be 

```
requireNonNull(type, "type is null")
```

File: lib/trino-geospatial-toolkit/src/main/java/io/trino/geospatial/serde/GeometrySerde.java [hunk 5]
```diff
@@ -81,32 +81,20 @@ public static GeometryType getGeometryType(Slice shape)
     private static void writeGeometry(DynamicSliceOutput output, OGCGeometry geometry)
     {
         GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
+
```
Comment (ebyhr) [inline lib/trino-geospatial-toolkit/src/main/java/io/trino/geospatial/serde/GeometrySerde.java:84]: nit: Revert unrelated change.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java [hunk 6]
```diff
@@ -1683,12 +1681,10 @@ public Optional<ConnectorTableExecuteHandle> getTableHandleForExecute(
             throw new IllegalArgumentException("Unknown procedure '" + procedureName + "'");
         }
 
-        switch (procedureId) {
-            case OPTIMIZE:
-                return getTableHandleForOptimize(tableHandle, executeProperties, retryMode);
-        }
-
-        throw new IllegalArgumentException("Unknown procedure: " + procedureId);
+        DeltaLakeTableHandle tableHandle = (DeltaLakeTableHandle) connectorTableHandle;
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:1684]: nit: Revert unrelated change.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 7]
```diff
@@ -392,30 +392,33 @@ public IcebergTableHandle getTableHandle(
...
-                            ? unpackMillisUtc((long) version.getVersion())
-                            : ((LongTimestampWithTimeZone) version.getVersion()).getEpochMillis();
-                }
-                else {
-                    throw new TrinoException(NOT_SUPPORTED, "Unsupported type for temporal table version: " + versionType.getDisplayName());
-                }
-                return getSnapshotIdAsOfTime(table, epochMillis);
+        return switch (version.getPointerType()) {
+            case TEMPORAL -> getTemporalSnapshotIdFromVersion(table, version, versionType);
+            case TARGET_ID -> getTargetSnapshotIdFromVersion(table, version, versionType);
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:397]: Optional: Could you add a preparatory commit to extract methods?

**Review by ebyhr — APPROVED** [review]

File: lib/trino-geospatial-toolkit/src/main/java/io/trino/geospatial/serde/GeometrySerde.java (hunk 5 above)
Comment (mx123) [reply lib/trino-geospatial-toolkit/src/main/java/io/trino/geospatial/serde/GeometrySerde.java]: done.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java (hunk 6 above)
Comment (mx123) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java]: done.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java (hunk 7 above)
Comment (mx123) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:397]: added: [Extract getting snapshotId from Iceberg metadata by types](https://github.com/trinodb/trino/pull/15395/commits/5f2ccea44af812f25f079b4e84c310cc853e4d9e)
